### DISCUSSION TITLE
How protect original urls to shorthened files ?

### DISCUSSION THREAD

**Discussion by conderus** [YOURLS/YOURLS#3104] [General]
Hi, I use YOURLS to shorten URLs to zip files on a website, but I would like to mask also these original URLs. I noticed that when a zip file is missing, then we have 404 of course but in the browser, we can see also the original path to this file. Also when I use some tools to see if the URL is still online, then it's possible to see in detail the path to the original files. Is it possible to mask these paths somehow? I want to return only zip to download, without the possibility to see original URL.

**Comment by dgw** [comment]
You want a fundamentally different tool.

By design, YOURLS is always going to send the client onward to the full URL. You're asking for, basically, a proxy that pulls the original file from where it lives and sends that in response to shortlink requests, instead.

It might be possible to implement that with a plugin, but it's likely better to spend that time on other things. [Token-based downloads](https://stackoverflow.com/q/65858875/5991), maybe (so even if someone shares the download link, it's useless without going through your download page and whatever authentication it requires again).

**Comment by conderus** [comment]
Thank you for your reply. I was looking also for some plugins, I thought it's easy to add such a feature. I will have to read more about it, because yes, such a token system is even much better, I want only certain people to be able to download these.
